fla/ops/common/backends/tilelang/chunk_bwd.py

```
@@ -168,23 +168,13 @@ def kernel_body(q, k, v, g, h, do, dh, dq, dk, dw, dv, dg, scale,
                     b_dk[_i, _j] * (T.exp2(-s_g[_i] + g_last) if _USE_EXP2 else T.exp(-s_g[_i] + g_last)),
                     0.0)
 
-            # dq*q and dk*k reductions: compute products directly in fragments
-            # via fragment×shared (safe per pattern 174). Avoids the shared
-            # round-trip through s_A1/s_A2.
-            f_prod1 = T.alloc_fragment((_BT, _BK), T.float32)
+            # dk*k reduction (before ds gemms): compute product directly in
+            # fragments via fragment×shared. b_dg_last uses the gated-only b_dk.
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)